### PR TITLE
Actually evaluate :face properties if appropriate

### LINKED ISSUE #153: Cannot create conditional face
Documentation about the `:face` segment property states:

> `:face`: The face in which to render the  segment.  This may be either a face or a form evaluating to a face. 

But the following code (from `emacs -q`) doesn't work:

``` emacs-lisp
(require 'spaceline-config)

(spaceline-compile 'thblt
		   '(("Hello" :face bold)
		     ("World" :face (if t bold italic)))
		   '())

(setq-default mode-line-format '("%e" (:eval (spaceline-ml-thblt))))
```

The mode line shows absolutely nothing, and evaluating the generated `spaceline-ml-thblt` raises an error:

```
Debugger entered--Lisp error: (error "Invalid face" if t bold italic)
  internal-get-lisp-face-attribute((if t bold italic) :inverse-video nil)
  face-attribute((if t bold italic) :inverse-video nil default)
  pl/background-color((if t bold italic))
  #[(face1 face2 &optional height) " ...
[...] [ Backtrace contains special characters I cannot paste in the GitHub issue.]
```

A quick look at Spaceline code seems to indicate that the documentation is actally wrong and that if :face is a form, it is not evaluated --- but I may be wrong, I'm very new to Elisp.

Thanks!

### PR DESCRIPTION
**Pull request by jyp**
Fixes #153

### CODE REVIEW

**Comment by jyp** [comment]
@thblt Ping, what about my later formulation?

**Comment by thblt** [comment]
@jyp  Sorry I forgot to answer (a bit too deep in real-life work these days...).  Your last formulation is extremely clear (I didn't understand before that only these 3 symbols got a special handling)

**Comment by jyp** [comment]
@thblt I have updated the documentation as we had agreed. Is this ok for merging?

**Comment by jyp** [comment]
@TheBB What about merging this?

**Comment by TheBB** [comment]
Yeah, in general I'm fine with it.

The original motivation for treating those ~~three~~ four symbols specially was to allow the user to just do `:face default-face` (which is actually a variable whose value is a face) or `:face some-actual-face` in the same manner, not needing to quote any of them.

With this PR, as is indicated in the doc changes, one would need to quote the second but not the first. That's fine, I think my original approach was a mistake and it's not good to have too complicated rules about what is evaluated and what isn't.

In summary, `face` is always then evaluated at runtime. Observe that if `face` is a list, `(,@face)` is the same thing as `face` (just imagine the backquote in there). So actually the two branches of this `if` evaluate to the same thing. I think then that we might as well drop it entirely.

**Comment by TheBB** [comment]
Thanks for the patch and sorry for the long wait.
